### PR TITLE
Bug 1583870 - Migrate build tasks for PRs and pushes

### PR DESCRIPTION
**Pull request by JohanLorenzo**
---

* ✅ tasks are green
* ✅ builds happen
* ✅ coverage reports are uploaded
* ✅ edge cases are taken care of 

### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/master/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### After merge
- [ ] **Milestone**: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- [ ] **Breaking Changes**: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### CODE REVIEW

File: taskcluster/ac_taskgraph/transforms/build.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    for task in tasks:
+        lint_task = 'lintRelease' if task.pop("lint-release") else 'lint'
+
+        for field in ('description', 'run.gradlew', 'treeherder.symbol'):
+            _format_recursively(task, field, component=task["name"], lint_task=lint_task)
+
+        yield task
+
+
+def _format_recursively(object, field, **format_kwargs):
```
Comment (mitchhentges) [inline taskcluster/ac_taskgraph/transforms/build.py]: This is called `_format_recursively`, but it doesn't recurse :thinking:

Comment (JohanLorenzo) [reply taskcluster/ac_taskgraph/transforms/build.py]: Yeah, I couldn't find a good name yesterday. I just changed it to `_deep_format()` which looks more accurate to me. Please let me know if you have a better idea 🙂
